### PR TITLE
bump OpenJPA version from 2.4.1 to 2.4.3 for Java 11 compliance

### PR DESCRIPTION
**Pull request by vorburger**
This will require # (which in turn first requires a few other minor PRs to be merged).

The upgrade is because 2.4.1 was not yet Java 11 compatible (due to an old ASM)
    
In the context of the Gradle upgrade work, I'd like to upgrade to Java 11 soon-ish, and am starting to experiment with it.

This first requires #604, #605, #606.

### CODE REVIEW

**Comment by vorburger** [comment]
This, ideally (to avoid doing required adjustment work twice) requires #607 as a pre-requisite, first.

**Comment by vorburger** [comment]
@conradsp what do you think?  @lingocoder FYI

**Comment by vorburger** [comment]
Actually, seeing how the merge of #607 apparently just broke #525 again, perhaps let's hold off merging this until affter #525 is merged, then rebase this on top of that, and only then merge this?

**Comment by vorburger** [comment]
> rebase this on top of that, and only then merge this?

done, and it passes. @awasum @conradsp @vishwasbabu would you like to merge this?

**Comment by vorburger** [comment]
FTR: https://issues.apache.org/jira/browse/OPENJPA-2730 has what we need OpenJPA v2.4.3. This alone doesn't quite do the trick for https://github.com/apache/fineract/pull/609, see https://github.com/radcortez/openjpa-gradle-plugin/issues/8 for why. IMHO it would still make sense to merge this already right now though; I'll deal with the follow-up in #609.

**Review by awasum — APPROVED** [review]

**Pull request merged**
